src/radau.jl

```diff
@@ -0,0 +1,351 @@
+        btab = bt_radau3
+    elseif stageNum ==5
+        btab = bt_radau5
+    elseif stageNum == 9
+        btab = bt_radau9
+    else
+        btab = constRadauTableau(stageNum)
+    end
+
+    ## previous data set to null to begin
+    tpre = NaN
+    ypre = zeros(y0)
+    dypre = zeros(y0)
+    step = 1
+    stageNum = stageNum
+    finished = false
+
+
+    ## intialize state
+    st =  RadauState{T,Y}(h,
+                   t, y, dy,
+                   tpre, ypre, dypre,
+                   step, finished,
+                   btab)
+
+    # Time stepping loop
+    while !done()
+        stats = trialstep!(st)
+        err, stats, st = errorcontrol!(st) # (1) adaptive stepsize (2) error
+        if err < 1
+            stats, st = ordercontrol!()
+            accept = true
+        else
+            rollback!()
+        end
+
+        return = status()
+    end
+end
+
+###########################################
+# iterator like helper functions
+###########################################
+
+function done(st)
+    @unpack st: h, t, tfinal
+    if h < minstep || t = tfinal
+        return true
+    else
+        return false
+    end
+end
+
+"trial step for ODE with mass matrix"
+function trialstep!(st)
+    @unpack st: h, t,y, tfinal
+
+
+    # Calculate simplified Jacobian if My' = f(t,y)
+    #     _                                             _
+    #    |M - h*a[11]*h*f(tn,yn) ... -h*a[1s]*f(tn,yn)   |
+    # G= |         ⋮              ⋱           ⋮           |
+    #    | - h*a[1s]*h*f(tn,yn) ...   M-h*a[ss]*f(tn,yn) |
+    #    |_                                             _|
+    #
+    g(z) = f(t,z)
+    J = ForwardDiff.jacobian(g, y)
+    I = eye(stageNum,stageNum)
+
+    #AoplusJ = [btab.a[i,j]*J for i=1:stageNum, j=1:stageNum]
+    AoplusJ=zeros(stageNum*dof,stageNum*dof)
+    for i=1:stageNum
+        for j = 1:stageNum
+            for l = 1:dof
+                for k = 1:dof
+                    AoplusJ[(i-1)*stageNum+l,(j-1)*stageNum+k] =btab.a[i,j]*J[l,k]
+                end
+            end
+        end
+    end
+
+    AoplusI = [btab.a[i,j]*I for i=1:stageNum, j=1:stageNum]
+    AoplusI2=zeros(stageNum*dof,stageNum*dof)
+    for i=1:stageNum
+        for j = 1:stageNum
+            for l = 1:dof
+                for k = 1:dof
+                    AoplusI2[(i-1)*stageNum+l,(j-1)*stageNum+k] =btab.a[i,j]*I[l,k]
+                end
+            end
+        end
+    end
+
+    #IoplusM = [I[i,j]*M for i=1:stageNum, j=1:stageNum]
+    IoplusM=zeros(stageNum*dof,stageNum*dof)
+    for i=1:stageNum
+        for j = 1:stageNum
+            for l = 1:dof
+                for k = 1:dof
+                    IoplusM[(i-1)*stageNum+l,(j-1)*stageNum+k] =I[i,j]*M[l,k]
+                end
+            end
+        end
+    end
+
+    G =  IoplusM-h*AoplusJ
+    Ginv = inv(G)
+
+    # Use Netwon interation
+    #
+    #   ⃗z^(k+1) = ⃗z ^ (k) - Δ⃗z^(k)
+    #TODO: use the transformation T^(-1)A^(-1)T = Λ, W^k = (T^-1 ⊕ I)Z^k
```

Comment:
Yes indeed, this is very very helpful!!! Saves me from having to implement tensor product myself. Thanks.